### PR TITLE
save and check the version of ONNX operation

### PR DESCRIPTION
**Pull request by chentong319**
This patch is intended to handle the change of ONNX specification. The current latest version of operations are saved as a dictionary in the script. The generation for onnx-mlir Tablegen will choose that version even if ONNX package is upgraded. 
To move forward to the latest version of onnx installed in your system, run the script with --generate-for-latest-version.  In addition to using the latest version for Tablegen, the change in version, compared to the saved version, is reported and the new version dictionary is generated on standard output.

### CODE REVIEW

File: utils/gen_doc.py [hunk 1]
```diff
@@ -29,12 +31,180 @@
                     help="Output OpBuildTable.inc content to stdout.",
                     action="store_true",
                     default=False)
+parser.add_argument("--generate-for-latest-version",
```
[the remark this answers is not in the document]
Comment (chentong319) [reply utils/gen_doc.py:34]: The latest version of your installation. This argument is only used by core developers. I am thinking of attaching the version number of onnx to the version_dict.

File: utils/gen_doc.py [hunk 2]
```diff
@@ -29,12 +31,180 @@
...
 parser.add_argument("--domain", 
                     help="specify domain, ONNX or ONNX_ML",
                     default = "ONNX")
 
 args = parser.parse_args()
 
+generate_for_latest_version = args.generate_for_latest_version
+
+
+#Record the version of each operation that is treated as the current version
```
[the remark this answers is not in the document]
Comment (chentong319) [reply utils/gen_doc.py:48]: Latest version of the installed ONNX. Will change the comment.

File: utils/CMakeLists.txt [hunk 3]
```diff
@@ -23,11 +23,11 @@ add_custom_target(OMONNXOpsIncTranslation
         DEPENDS OMONNXOpsTableGenIncGen
                 OMONNXOpsBuildTableIncGen)
 
-# Invoke gen_doc.py to obtain ONNXOps.td.inc, OpBuildTable.inc.
+# Invoke gen_onnx_mlir.py to obtain ONNXOps.td.inc, OpBuildTable.inc.
```
Comment (tjingrant) [inline utils/CMakeLists.txt:26]: MLONNXOps.td.inc? MLOpBuildTable.inc?

Comment (chentong319) [reply utils/CMakeLists.txt:26]: I will fix these minor errors in my next patch.

**Pull request merged**